packages/@aws-cdk/aws-codepipeline/test/notifications.test.ts

```diff
@@ -0,0 +1,206 @@
+import '@aws-cdk/assert-internal/jest';
+import * as notifications from '@aws-cdk/aws-codestarnotifications';
+import * as cdk from '@aws-cdk/core';
+import * as codepipeline from '../lib';
+import { FakeBuildAction } from './fake-build-action';
+import { FakeSourceAction } from './fake-source-action';
+
+describe('pipeline with codestar notification integration', () => {
+  let stack: cdk.Stack;
+  let pipeline: codepipeline.Pipeline;
+  let sourceArtifact: codepipeline.Artifact;
+  beforeEach(() => {
+    stack = new cdk.Stack();
+    sourceArtifact = new codepipeline.Artifact();
+    pipeline = new codepipeline.Pipeline(stack, 'Pipeline', {
+      stages: [
+        {
+          stageName: 'Source',
+          actions: [new FakeSourceAction({ actionName: 'Fetch', output: sourceArtifact })],
+        },
+        {
+          stageName: 'Build',
+          actions: [new FakeBuildAction({ actionName: 'Gcc', input: sourceArtifact })],
+        },
+      ],
+    });
+  });
+
+  test('On "Action state change" events emitted notification rule', () => {
+    pipeline.notifyOnActionStateChange('NotifyOnActionStateChange', { target: { bind: () => ({ targetType: notifications.TargetType.AWS_CHATBOT_SLACK, targetAddress: 'SlackID' }) } });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'PipelineNotifyOnActionStateChange5148B236',
+      DetailType: 'FULL',
+      EventTypeIds: [
+        'codepipeline-pipeline-action-execution-succeeded',
+        'codepipeline-pipeline-action-execution-failed',
+        'codepipeline-pipeline-action-execution-canceled',
+        'codepipeline-pipeline-action-execution-started',
+      ],
+      Resource: {
+        'Fn::Join': [
+          '',
+          [
+            'arn:',
+            {
+              Ref: 'AWS::Partition',
+            },
+            ':codepipeline:',
+            {
+              Ref: 'AWS::Region',
+            },
+            ':',
+            {
+              Ref: 'AWS::AccountId',
+            },
+            ':',
+            {
+              Ref: 'PipelineC660917D',
+            },
+          ],
+        ],
+      },
```

Comment:
Reformat these so that they take less space:

```suggestion
        'Fn::Join': ['', [
          'arn:',
          { Ref: 'AWS::Partition' },
          ':codepipeline:',
          { Ref: 'AWS::Region' },
          ':',
          { Ref: 'AWS::AccountId' },
          ':',
          { Ref: 'PipelineC660917D' },
        ]],
      },
```

Same comment in other tests.

Reply:
Already adjusted.